sdk/core/azure-core-amqp/test/ut/link_tests.cpp

```diff
@@ -45,9 +45,10 @@ namespace Azure { namespace Core { namespace Amqp { namespace Tests {
       Link link1(session, "MySession", SessionRole::Sender, "Source1", "Target1");
       Link link2(session, "MySession", SessionRole::Sender, "Source2", "Target2");
     }
-
+#if ENABLE_UAMQP
     GTEST_LOG_(INFO) << LinkState::Error << LinkState::Invalid << static_cast<LinkState>(92)
                      << LinkState::HalfAttachedAttachReceived;
+#endif
```

Comment:
I am unable to follow/understand why we end up needing the #ifs in some of these tests, in these particular places. Can you provide some context on how you went about introducing these?

Reply:
The logic is relatively straightforward:

Code which is uAMQP only is marked as uAMQP only.
Code which is Rust AMQP only is marked as Rust AMQP only is marked as Rust AMQP only.

In this case, the LinkState enumeration is uAMQP only.